src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestCreationParameters.cs

```diff
@@ -0,0 +1,129 @@
+        /// </summary>
+        /// <remarks>https://tools.ietf.org/html/draft-ietf-oauth-signed-http-request-03#section-3</remarks> 
+        public bool CreateM { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the <see cref="SignedHttpRequestClaimTypes.U"/> claim should be created and added or not.
+        /// </summary>
+        /// <remarks>https://tools.ietf.org/html/draft-ietf-oauth-signed-http-request-03#section-3</remarks> 
+        public bool CreateU { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the <see cref="SignedHttpRequestClaimTypes.P"/> claim should be created and added or not.
+        /// </summary>
+        /// <remarks>https://tools.ietf.org/html/draft-ietf-oauth-signed-http-request-03#section-3</remarks>
+        public bool CreateP { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the <see cref="SignedHttpRequestClaimTypes.Q"/> claim should be created and added or not.
+        /// </summary>
+        /// <remarks>https://tools.ietf.org/html/draft-ietf-oauth-signed-http-request-03#section-3</remarks> 
+        public bool CreateQ { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the <see cref="SignedHttpRequestClaimTypes.H"/> claim should be created and added or not.
+        /// </summary>
+        /// <remarks>https://tools.ietf.org/html/draft-ietf-oauth-signed-http-request-03#section-3</remarks> 
+        public bool CreateH { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the <see cref="SignedHttpRequestClaimTypes.B"/> claim should be created and added or not.
+        /// </summary>
+        /// <remarks>https://tools.ietf.org/html/draft-ietf-oauth-signed-http-request-03#section-3</remarks> 
+        public bool CreateB { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets the <see cref="AdditionalClaimCreator"/> delegate.
+        /// </summary>
+        public AdditionalClaimCreator AdditionalClaimCreator { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="CustomNonceCreator"/> delegate. 
+        /// </summary>
+        public CustomNonceCreator CustomNonceCreator { get; set; }
+
+        /// <summary>
+        /// Default value for the <see cref="TimeAdjustment"/>.
+        /// </summary>
+        public static readonly TimeSpan DefaultTimeAdjustment = TimeSpan.Zero;
+
+        /// <summary>
+        /// Gets or sets a time adjustment to apply when creating the timestamp ("ts") claim.
+        /// </summary>
+        /// <remarks>Allows for adjusting the local time so it matches a server time.</remarks>
+        public TimeSpan TimeAdjustment { get; set; } = DefaultTimeAdjustment;
```

Comment:
Should we allow this to be negative?

Reply:
I think that there is no harm in allowing negative values, thought I can't think of a use case where it might be applicable.